### ISSUE TITLE
Scene File for San Miguel

### ISSUE THREAD

**Issue by knightcrawler25** [Opioid/sprout#2]
Do you happen to have the scene file for the San Miguel 2.0 scene as in the cover pic in the description? Would be helpful as a sample :)

**Comment by Opioid** [comment]
If you click on the link below the image you will be directed to a list of freely available scenes, including San Miguel 2.0.

**Comment by knightcrawler25** [comment]
Sorry for not being clear. I meant the scene description for that particular scene used by your renderer.

**Comment by Opioid** [comment]
Ah, yes. I'll get back to you about that, tomorrow or so.

**Comment by knightcrawler25** [comment]
Sure. No problem :)

**Comment by Opioid** [comment]
https://drive.google.com/file/d/1ZShkplL3RqLGDsx5ySag31PwfEGLcDD2/view?usp=sharing

I hope this package includes everything you need. 

Personally I always use a data structure like this:
/
    data/
        takes/...
        scenes/...
    system/
        sprout-executable

Then you should be able to render the scene from system: ./sprout -i "takes/san_miguel.json"
Let me know how it goes.

**Comment by knightcrawler25** [comment]
Yes that file has all the things I needed. Tried rendering the scene and it works. :D Thanks a lot :) Beautiful project you've got here.

**Comment by Opioid** [comment]
Thanks for the feedback!

**Comment by knightcrawler25** [comment]
I have another question. How are the material conversions being done from the .mtl file to the scene file? Is it mostly manual tweaking until it looks right?

**Comment by Opioid** [comment]
Well, the tool I use to convert data to a format the renderer understands, is based on [assimp](http://www.assimp.org/) and can extract at least some useful material information from the mtl. It is in a kind of messy state at the moment as it is a modified leftover from a different [project](https://github.com/Opioid/Substitute). Hopefully I get around to clean it up at some point. 

I use the following formula to convert from mtl shininess to roughness:
```
float shininess_to_roughness(float shininess) {
    return std::pow(2.f / (shininess + 2.f), 0.25f);
}
```
But, to answer your original question: Yes, I actually re-touched most materials or made them metallic and so on. For more complex scenes this workflow can be become a bit tedious I'm afraid.

**Comment by knightcrawler25** [comment]
That seems like a lot of hard work to modify the materials for each scene. Also a suggestion: Maybe you can add the .scene and .take files of the scenes that you've rendered (in the gallery) to the repository as well. The meshes and textures are downloadable from the links and it would be easy to modify a scene to experiment than write it from scratch again, especially for the larger scenes. Thanks for all the help so far :)

**Comment by Opioid** [comment]
I agree, tweaking the materials is a lot of work and it would generally be useful to be able to quickly test scenes in different formats. 
Nevertheless, I think I'll keep data conversion out of the renderer project, but maybe I'll improve the conversion tool at some point.
For this reason it would still be some work to recreate the images, even with the original source data and my take/scene files.
I'm happy to help anyone who is interested in testing the renderer, though.

**Comment by knightcrawler25** [comment]
Ah I see. Makes sense since one would have to convert the data to json and also modify the texture formats and so on. Also, I noticed yesterday that the renderer crashed twice at frame 97 for the scene you provided. I'll try and see if I can reproduce it when I get home and provide you with more details

**Comment by Opioid** [comment]
Exactly, the content pipeline leaves a lot to be desired. But for now, it is what it is.

Aha, so you are rendering multiple frames. Did you enable the moving sun? I forgot I had it in the scene.
Does the crash only occur if you render a sequence from 0 to 97+ or also if you just render frame 97? The latter would make debugging a lot easier of course.

If you want to experiment, you could try to make a build with the NaN option turned on in cmake. In the worst case you might get spammed with information in case I broke something since last rendering that scene. But sometimes it gives useful hints.

Thanks for reporting the crash! I'll look into it sometime.

**Comment by PotatoPot6676** [comment]
Hey, Just came across your project. You have a really amazing renderer. Quite a huge project for a single person to work on. Very Impressive! I downloaded and ran the san miguel file that you've shared above and it works fine but I've gone through the gallery and saw the rendering of the Lumberyard scene. The rendering looks very beautiful in your gallery and I would like really like to try and render it on my PC with your renderer and play around with the scene. Could you share the file for that if it's not too much trouble?

**Comment by Opioid** [comment]
Thanks for the feedback! 

Try this:
https://drive.google.com/file/d/1b2AyTGEfRr1KDVx2gKcqkyeZiKKwadHf/view?usp=sharing
It is pretty big, because it includes all the textures that come with the original package.

**Comment by PotatoPot6676** [comment]
Thanks a lot!! It works :+1:

**Comment by PotatoPot6676** [comment]
Is the progressive mode option not working as of now? I tried the -p arg and it crashes after loading the data, just before the rendering starts

**Comment by Opioid** [comment]
Yeah, it doesn't work out of the box. If you are interested, I can explain how to create a working build. But I'm not really giving it any attention myself.
